packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,709 @@
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+*data.js:*
+```js
+export default [{
+    ID: 1,
+    CompanyName: 'Super Mart of the West',
+    Address: '702 SW 8th Street',
+    City: 'Bentonville',
+    State: 'Arkansas',
+    Zipcode: 72716
+  }, {
+    ID: 2,
+    CompanyName: 'Electronics Depot',
+    Address: '2455 Paces Ferry Road NW',
+    City: 'Atlanta',
+    State: 'Georgia',
+    Zipcode: 30339
+  }, {
+    ID: 3,
+    CompanyName: 'K&S Music',
+    Address: '1000 Nicllet Mall',
+    City: 'Minneapolis',
+    State: 'Minnesota',
+    Zipcode: 55403
+  }, {
+    ID: 4,
+    CompanyName: "Tom's Club",
+    Address: '999 Lake Drive',
+    City: 'Issaquah',
+    State: 'Washington',
+    Zipcode: 98027
+  }, {
+    ID: 5,
+    CompanyName: 'E-Mart',
+    Address: '3333 Beverly Rd',
+    City: 'Hoffman Estates',
+    State: 'Illinois',
+    Zipcode: 60179
+  }];
+```
+
+Take a look at the image below to see what you will get by the end of the tutorial. At the top right corner, you can see a select box. When a user selects a value from it, the grid displays only the filtered rows. When the value is cleared using the `CLEAR` button, the grid displays all rows.
+
+<p align="center">
+  <img class="img-responsive" src="../../../../../img/guides/custom-plugin-development/result.png">
+</p>
+
+The first step is to render custom markup.
+
+## Render Custom Markup
+
+To render custom markup in a UI element that belongs to the Grid ([toolbar](../reference/toolbar.md) in our case), we need to create a plugin. Wrap the custom markup in the [Plugin](../../../core/docs/reference/plugin.md) component to do this. We will name the plugin `ToolbarFilter`, since it will be part of the toolbar and will contain a filtering UI element.
+
+Create a new folder called `plugins` in the project and add a new file, `toolbar-filter.js`, with the following code to it.
+
+*toolbar-filter.js:*
+```jsx
+import {
+  Plugin
+} from '@devexpress/dx-react-core';
+
+export class ToolbarFilter extends React.PureComponent {
+  render() {
+    return (
+      <Plugin
+        name="ToolbarFilter"
+      >
+      </Plugin>
+    );
+  }
+}
+```
+
+The Grid's toolbar exports a [template](../../../core/docs/reference/template.md) with the `toolbarContent` name. Add the [Template](../../../core/docs/reference/template.md) component to the plugin and set its `name` attribute to `toolbarContent`. This will be the place for our custom markup.
+
+```jsx
+...
+import {
+  ...,
+  Template
+} from '@devexpress/dx-react-core';
+
+export class ToolbarFilter extends React.PureComponent {
+  render() {
+    return (
+      <Plugin
+        ...
+      >
+        <Template name="toolbarContent" />
+      </Plugin>
+    );
+  }
+}
+```
+
+We do not want to *replace* the default toolbar content. Rather we should *append* our content to it. To do this, add the [TemplatePlaceholder](../../../core/docs/reference/template-placeholder.md) component without parameters to the template:
```

Comment:
"We do not want to *replace* the default toolbar content. Rather we should *append* our content to it." -> We should *append* our content to default toolbar.